Library/Homebrew/os/linux/elf.rb

```diff
@@ -47,12 +47,14 @@ def initialize(path)
         @dylibs = []
         return
       end
-      command = ["ldd", path.expand_path.to_s]
-      libs = Utils.popen_read(*command).split("\n")
-      raise ErrorDuringExecution, command unless $?.success?
-      needed << "not found"
-      libs.select! { |lib| needed.any? { |soname| lib.include? soname } }
-      @dylibs = libs.map { |lib| lib[LDD_RX, 1] || lib[LDD_RX, 2] }.compact
+      if path.x86_64?
```

Comment:
I've created PR https://github.com/Linuxbrew/brew/pull/370 to add `Hardware::CPU.arch`.
I suggest
```diff
-if needed.empty?
+if needed.empty? || path.arch != Hardware::CPU::arch
   @dylibs = []
   return
 end
```